### PR TITLE
task14

### PR DESCRIPTION
**Pull request by FunkyColdMedyna**
Hi all please checkout our work for task 14.  
Stephanie and I created the basics for our user schema.  

we left a userId section commented out. We couldn't recall if that was something that would be generated through user registration.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -20,8 +20,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-router": "^6.4.2",
-    "react-router-dom": "^6.4.2",
-
+    "react-router-dom": "^6.4.2"
```
Comment (FunkyColdMedyna) [inline client/package.json:23]: removed unnecessary comma (this change was made in previous commit webpack config changes but hasnt been merged yet)

File: client/src/components/imageCollage.js [hunk 2]
```diff
@@ -34,8 +34,8 @@ export default function IamgeCollage() {
 
 const itemData = [
   {
-    img: 'https://images.unsplash.com/photo-1551963831-b3b1ca40c98e',
-    title: 'Breakfast',
+    img: 'http://placekitten.com/g/400/500',
+    title: 'placekitten',
```
Comment (FunkyColdMedyna) [inline client/src/components/imageCollage.js:38]: testing different picture sizes within image collage

File: server/models/User.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        required: [true, 'Please provide a first name']
+    },
+    lastName: {
+        type: String,
+        required: [true, 'please provide a last name']
+    },
+    gender:{
+        type: String,
+        required: [true, 'please select your gender'],
+        enum: [
```
Comment (hjkmines) [inline server/models/User.js:31]: should be `required` not `require`

**Review by hjkmines — APPROVED** [review]
LGTM!

**Review by jkyan1988 — APPROVED** [review]
Looks good.

File: server/models/User.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const UserSchema = new Schema({
+    userName: {
+        type: String,
+        unique: true,
+        required: [true, 'Please provide a user name'],
+        maxLength: [12, 'User name cannot exceed 12 characters']
+    },
+    password:{
+        type: String,
+        required: [true, 'please provide a password'],
```
Comment (vmaineng) [inline server/models/User.js:13]: Just curious, for password and email, should we add in validation?

Comment (FunkyColdMedyna) [reply server/models/User.js:13]: we can do some regex to make sure that there are the generally accepted requirements (at least one lower case, one upper case letter, one number, etc.) Not sure if we want to do that in the schema or if we want to nest that into the registration for the user

Comment (vmaineng) [reply server/models/User.js:13]: Thanks for letting me know, Chris.

**Review by vmaineng — APPROVED** [review]

**Pull request merged**